backend/asmpackager.ml

```diff
@@ -204,77 +212,58 @@ let build_package_cmx members cmxfile =
       (fun m accu ->
         match m.pm_kind with PM_intf -> accu | PM_impl info -> info :: accu)
       members [] in
-  (* CR vlaviron: The code for [pack_units1] and [pack_units2] relies on
-     [Compilenv.unit_for_global] to find the correct compilation unit for the
-     packed modules. This is fragile, and means that we must keep
-     [Compilenv.global_infos_table] alive longer than we would otherwise need.
-     We should consider computing the sets earlier, to remove the dependency
-     on the table. *)
-  let pack_units1 : Compilation_unit.Set.t lazy_t =
-    lazy (List.fold_left
-            (fun set info ->
-               let unit_id = Compilenv.unit_id_from_name info.ui_name in
-               Compilation_unit.Set.add
-                 (Compilenv.unit_for_global unit_id) set)
-            Compilation_unit.Set.empty units)
+  let pack_units : Compilation_unit.Set.t lazy_t =
+    lazy (List.map (fun info -> info.ui_name) units
+            |> Compilation_unit.Set.of_list)
   in
-  let pack_units2 : Flambda2_identifiers.Compilation_unit.Set.t lazy_t =
-    let unit_for_global ident : Flambda2_identifiers.Compilation_unit.t =
-      let linkage_name : Flambda2_identifiers.Linkage_name.t =
-        ident
-        |> Compilenv.unit_for_global
-        |> Compilation_unit.get_linkage_name
-        |> Linkage_name.to_string
-        |> Flambda2_identifiers.Linkage_name.create
-      in
-      Flambda2_identifiers.Compilation_unit.create ~name:(Ident.name ident)
-        linkage_name
-    in
-    lazy (List.fold_left
-            (fun set info ->
-               let unit_id : Ident.t =
-                 Compilenv.unit_id_from_name info.ui_name
-               in
-               Flambda2_identifiers.Compilation_unit.Set.add
-                 (unit_for_global unit_id) set)
-            Flambda2_identifiers.Compilation_unit.Set.empty units)
+  let ui = Compilenv.current_unit_infos() in
+  (* XXX What happens if we have -for-pack Foo.Bar? *)
+  (* XXX Should we assert something about ui.ui_name -- e.g. no prefix? *)
```

Comment:
These XXX comments and various debugging code need removing.  I think @lpw25 should read the diff to this file.